setting_list.c

```diff
@@ -173,11 +183,85 @@ void setting_get_string_representation_size_in_mb(void *data,
             (*setting->value.target.sizet)/(1024*1024));
 }
 
-static int setting_uint_action_left_default(void *data, bool wraparound)
+void setting_get_string_representation_uint_as_enum(void *data,
+      char *s, size_t len)
+{
+   rarch_setting_t *setting = (rarch_setting_t*)data;
+   if (setting)
+      snprintf(s, len, "%s",
+            msg_hash_to_str(setting->index_offset+(*setting->value.target.unsigned_integer)));
+}
+
+static float recalc_step_based_on_length_of_action(rarch_setting_t *setting)
+{
+   float step = setting->step;
+#ifdef HAVE_MENU
+   global_t *global = global_get_ptr();
+   if ( global )
+   {
+#if 0
+      if (setting->enforce_minrange)
+      {
+         float numsteps = (setting->max-setting->min)/setting->step ;
+         float multiplier = 1.0f ;
+         if ( global->menu.action_press_time  > 12000000)
+         {
+            multiplier = (numsteps/60.0f);
+         } else if ( global->menu.action_press_time  > 9000000)
+         {
+            multiplier = (numsteps/300.0f) ;
+         } else if ( global->menu.action_press_time  > 6000000)
+         {
+            multiplier = (numsteps/750.0f);
+         } else if ( global->menu.action_press_time  > 3000000)
+         {
+            multiplier = (numsteps/3000.0f) ;
+         } else
+         {
+            multiplier = 1.0f ;
+         }
+         step = setting->step*multiplier;
+      }
+      else
+#endif
+      {
+         if ( global->menu.action_press_time  > 21000000)
+         {
+            step = setting->step*1000000.0f ;
+         } else if ( global->menu.action_press_time  > 18000000)
+         {
+            step = setting->step*10000.0f ;
+         } else if ( global->menu.action_press_time  > 15000000)
+         {
+            step = setting->step*1000.0f ;
+         } else if ( global->menu.action_press_time  > 12000000)
+         {
+            step = setting->step*100.0f ;
+         } else if ( global->menu.action_press_time  > 9000000)
+         {
+            step = setting->step*100.0f ;
```

Comment:
Looks like the 12000000 and 9000000 conditions do the same thing?

In addition, they seem like a lot of [magic numbers](https://en.wikipedia.org/wiki/Magic_number_%28programming%29). Could there be a way to clean up their use? If not, that's understandable.